backend/pkg/connect/get_connectors.go

```diff
@@ -28,7 +29,25 @@ const (
 	connectorStateUnassigned connectorState = "UNASSIGNED"
 	connectorStateRunning    connectorState = "RUNNING"
 	connectorStatePaused     connectorState = "PAUSED"
-	connectorStateFAILED     connectorState = "FAILED"
+	connectorStateFailed     connectorState = "FAILED"
+	connectorStateRestarting connectorState = "RESTARTING"
+)
+
+// connectorStatus is our holistic unified connector status that takes into account not just the
+// connector instance state, but also state of all the tasks within the connector
+type connectorStatus = string
+
+const (
+	// Connector is in "running" state, >0 tasks, all of them running state
+	connectorStatusHealthy connectorStatus = "HEALTHY"
+	// Connector is "error" state
+	connectorStatusUnhealthy connectorStatus = "UNHEALTHY"
+	// Connector is "running" state, 0 tasks OR at least one task in failed state
+	connectorStatusDegraded connectorStatus = "DEGRADED"
```

Comment:
I think if it has 0 (running) tasks we know that the connector will not produce any data at all and hence should be in state unhealthy. @vladoschreiner can confirm